middle_end/flambda2/simplify_shared/closure_offsets.ml

```diff
@@ -817,21 +815,30 @@ module Greedy = struct
     used_closure_vars, offsets
 end
 
-type t = Greedy.state
+type t = Set_of_closures.t list
+
+let print fmt l =
+  Format.fprintf fmt "@[<hv>%a@]" (Format.pp_print_list Set_of_closures.print) l
 
-let print = Greedy.print
+let create () = []
 
-let create () = Greedy.create_initial_state ()
+let add_set_of_closures l ~is_phantom ~all_code:_ set_of_closures =
+  if is_phantom then l else set_of_closures :: l
 
-let add_set_of_closures state ~is_phantom ~all_code set_of_closures =
-  Greedy.create_slots_for_set ~is_phantom state all_code set_of_closures
+let add_offsets_from_function l1 ~from_function:l2 = l2 @ l1
```

Comment:
We could use here a `List.rev_append`, since we do not really case about the order of the list.

Reply:
Done